tests/ml/workflow/test_train_bandit.py

```diff
@@ -1 +1,185 @@
+            "y_test": y[offset:],
+        }
+
+        # country as id list dataset
+        cls.data_country_id_list = train_bandit.preprocess_data(
+            raw_data, cls.experiment_specific_params_country_as_id_list
+        )
+        X, y = train_bandit.data_to_pytorch(
+            cls.data_country_id_list,
+            cls.experiment_specific_params_country_as_id_list["features"],
+            cls.experiment_specific_params_country_as_id_list["product_sets"],
+        )
+        cls.X_country_id_list = {
+            "X_train": {
+                "X_float": X["X_float"][:offset],
+                "X_id_list": X["X_id_list"][:offset],
+                "X_id_list_idxs": X["X_id_list_idxs"][:offset],
+            },
+            "y_train": y[:offset],
+            "X_test": {
+                "X_float": X["X_float"][offset:],
+                "X_id_list": X["X_id_list"][offset:],
+                "X_id_list_idxs": X["X_id_list_idxs"][offset:],
+            },
+            "y_test": y[offset:],
+        }
+
+        # train benchmark models
+        cls.results_gbdt = benchmarks.fit_sklearn_gbdt(
+            X_train=cls.X_country_categ["X_train"]["X_float"],
+            y_train=cls.X_country_categ["y_train"],
+            X_test=cls.X_country_categ["X_test"]["X_float"],
+            y_test=cls.X_country_categ["y_test"],
+            hyperparams=cls.shared_params,
+        )
+
+        cls.results_mlp = benchmarks.fit_sklearn_mlp(
+            X_train=cls.X_country_categ["X_train"]["X_float"],
+            y_train=cls.X_country_categ["y_train"],
+            X_test=cls.X_country_categ["X_test"]["X_float"],
+            y_test=cls.X_country_categ["y_test"],
+            hyperparams=cls.shared_params,
+        )
+
+    @classmethod
+    def tearDownClass(cls):
+        pass
+
+    def test_pytorch_model_country_as_categorical(self):
+        pytorch_net = train_bandit.build_pytorch_net(
+            feature_specs=self.experiment_specific_params_country_as_categorical[
+                "features"
+            ],
+            product_sets=self.experiment_specific_params_country_as_categorical[
+                "product_sets"
+            ],
+            layers=self.shared_params["model"]["layers"],
+            activations=self.shared_params["model"]["activations"],
+            input_dim=train_bandit.num_float_dim(self.data_country_categ),
+        )
+
+        skorch_net = train_bandit.fit_custom_pytorch_module_w_skorch(
+            module=pytorch_net,
+            X=self.X_country_categ["X_train"],
+            y=self.X_country_categ["y_train"],
+            hyperparams=self.shared_params,
+        )
+
+        train_mse = skorch_net.history[-1]["train_loss"]
+        test_mse = skorch_net.history[-1]["valid_loss"]
+
+        # make sure mse is better or close to out of the box GBDT
+        assert train_mse < self.results_gbdt["mse_train"] * 1.05
+        assert test_mse < self.results_gbdt["mse_test"] * 1.05
```

Comment:
when making tests against MSE, is there any risk of flakiness?

Reply:
definitely sum, i'm hoping with enough epochs (50) and the 1.05 buffer we should be good